src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -640,15 +632,49 @@ def cancel_job(self, job_id: str, parent_job_id: str = None) -> None:
         then attempts to cancel it.
         If either of those steps fail, a NarrativeException is raised.
         """
+        checked_jobs = self._check_job_list([job_id])
 
-        if job_id is None or job_id == "":
-            raise ValueError("Job id required for cancellation!")
-        if not parent_job_id and job_id not in self._running_jobs:
+        if len(checked_jobs["error"]):
             raise ValueError(f"No job present with id {job_id}")
 
+        # otherwise, our job ID is fine
         if job_id in self._completed_job_states:
-            return
+            return True
+
+        if not self._check_job_terminated(job_id):
+            return self._cancel_job(job_id, parent_job_id)
+
+        return True
+
+    def cancel_jobs(
+        self, job_id_list: List[str] = [], parent_job_id: str = None
+    ) -> None:
+        """
+        Cancel a list of jobs
+
+        Results are returned as a dict of job status objects keyed by job id
 
+        :param job_id_list: list of strs
+        :return job_states: dict with keys job IDs and values job state objects
+
+        """
+        checked_jobs = self._check_job_list(job_id_list)
+
+        for job_id in checked_jobs["job_id_list"]:
+            if (
+                job_id not in self._completed_job_states
+                and not self._check_job_terminated(job_id)
+            ):
+                self._cancel_job(job_id)
+
+        job_states = self._construct_job_status_set(checked_jobs["job_id_list"])
+
+        for job_id in checked_jobs["error"]:
+            job_states[job_id] = {"job_id": job_id, "status": "does_not_exist"}
```

Comment:
`cancel_job` and `cancel_jobs` return job states, so if the job ID does not exist, return a job state indicating that the job could not be found.